pkg/server/local_observer.go

```diff
@@ -315,12 +315,13 @@ type flowsReader struct {
 // newFlowsReader creates a new flowsReader that uses the given RingReader to
 // read through the ring buffer. Only flows that match the request criterias
 // are returned.
-func newFlowsReader(r *container.RingReader, req *observer.GetFlowsRequest, log *logrus.Entry) (*flowsReader, error) {
-	whitelist, err := filters.BuildFilterList(req.Whitelist)
+func (s *LocalObserverServer) newFlowsReader(ctx context.Context, r *container.RingReader, req *observer.GetFlowsRequest, log *logrus.Entry) (*flowsReader, error) {
```

Comment:
Same comment as for `newRingReader` about the context passed to this constructor. I'm also a bit concerned about this constructor becoming a `LocalObserverServer` method. Isn't it possible to avoid this?

Reply:
Fixed as discussed on Slack